File: my-folder/0747-min-cost-climbing-stairs/solution.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n = cost.size();
        int size = n+2;
        int dp[size];
        memset(dp, 0, sizeof(dp));

        for(int i = n-1; i >=0; i--){
            dp[i] = cost[i] + min(dp[i+1], dp[i+2]);
        }
        return min(dp[0], dp[1]);
    }
};

File: my-folder/0747-min-cost-climbing-stairs/solution.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        c0, c1=cost[0], cost[1]
        for i in range(2, len(cost)):
            c0, c1=c1, min(c0, c1)+cost[i]
        return min(c0, c1)
